controllers/metal3datatemplate_controller.go

```diff
@@ -110,14 +107,16 @@ func (r *Metal3DataTemplateReconciler) Reconcile(req ctrl.Request) (_ ctrl.Resul
 		return ctrl.Result{}, errors.Wrapf(err, "failed to create helper for managing the metadata")
 	}
 
-	metadataLog = metadataLog.WithValues("cluster", cluster.Name)
-	if err := metadataMgr.SetClusterOwnerRef(cluster); err != nil {
-		return ctrl.Result{}, err
-	}
-	// Return early if the Metadata or Cluster is paused.
-	if util.IsPaused(cluster, capm3DataTemplate) {
-		metadataLog.Info("reconciliation is paused for this object")
-		return ctrl.Result{Requeue: true, RequeueAfter: requeueAfter}, nil
+	if capm3DataTemplate.Spec.ClusterName != "" && cluster.Name != "" {
```

Comment:
Why are we not checking this `cluster != nil` condition here?

Reply:
because the cluster is already defined, just with empty values, so cluster would never be nil